internalshared/configutil/kms.go

```diff
@@ -438,6 +441,31 @@ func getEnvConfig(kms *KMS) map[string]string {
 	return envValues
 }
 
+func mergeTransitConfig(config map[string]string, envConfig map[string]string) {
+	useFileTlsConfig := false
+	for _, varName := range TransitTlsConfigVars {
+		if _, ok := config[varName]; ok {
+			useFileTlsConfig = true
+			break
+		}
+	}
+
+	if useFileTlsConfig {
+		for _, varName := range TransitTlsConfigVars {
+			delete(envConfig, varName)
+		}
+	}
+
+	for varName, val := range envConfig {
+		// for some values, file config takes precedence
+		if slices.Contains(TransitPrioritizeConfigValues, varName) && config[varName] != "" {
```

Comment:
Can we use `strutil.StrListContains` from [go-secure-stdlib](https://github.com/hashicorp/go-secure-stdlib/blob/main/strutil/strutil.go#L29C6-L29C21) instead of adding a dependency to `k8s.io/utils/strings/slices`